backend/amp-live-list.go

```diff
@@ -157,13 +165,71 @@ func createMetadata(r *http.Request) []BlogPosting {
 	return result
 }
 
-func createLiveBlogSample(newStatus int, timestamp time.Time, r *http.Request) LiveBlogSample {
+func createLiveBlogSample(newStatus int, timestamp time.Time, r *http.Request, page Page) LiveBlogSample {
 	if newStatus > len(blogs) {
 		newStatus = len(blogs)
 	}
 	blogItems := getBlogEntries(newStatus, timestamp)
 	score := createScore(newStatus, 0)
-	return LiveBlogSample{BlogItems: blogItems, FootballScore: score, BlogMetadata: createMetadata(r)}
+	firstBlogID := strings.TrimPrefix(r.URL.Query().Get("from"), BLOG_ID_PREFIX)
+	firstItemIndex := getBlogEntryIndexFromID(firstBlogID, blogItems)
+	lenghtCurrentPageBlog := int(math.Min(float64(len(blogItems)), float64(firstItemIndex+MAX_BLOG_ITEMS_NUMBER_PER_PAGE)))
+
+	return LiveBlogSample{BlogItems: blogItems[firstItemIndex:lenghtCurrentPageBlog],
+		FootballScore: score,
+		BlogMetadata:  createMetadata(r),
+		NextPageURL:   getNextPageUrl(firstItemIndex, r, page, blogItems),
+		PrevPageURL:   getPrevPageUrl(firstItemIndex, r, page, newStatus),
+		PageNumber:    getPageNumberFromProductIndex(firstItemIndex)}
+}
+
+func getNextPageUrl(firstItemIndex int, r *http.Request, page Page, blogItems []BlogItem) string {
+	nextPageFirstItemIndex := firstItemIndex + MAX_BLOG_ITEMS_NUMBER_PER_PAGE
+	nextPageId := getNextPageId(r, page, blogItems, nextPageFirstItemIndex)
+	if nextPageId != "" {
+		return buildPaginationURL(r, page, nextPageId)
+	}
+	return ""
+}
+
+func getPrevPageUrl(firstItemIndex int, r *http.Request, page Page, newStatus int) string {
+	prevPageId := getPrevPageId(firstItemIndex, r, page, newStatus)
+	if prevPageId != "" {
+		return buildPaginationURL(r, page, prevPageId)
+	}
+	return ""
+}
+
+func getNextPageId(r *http.Request, page Page, blogItems []BlogItem, nextPageFirstItemIndex int) string {
+	if nextPageFirstItemIndex < len(blogItems) {
+		return fmt.Sprintf("%s%d", BLOG_ID_PREFIX, nextPageFirstItemIndex+1)
+	}
+	return ""
+}
+
+func getPrevPageId(firstItemIndex int, r *http.Request, page Page, newStatus int) string {
+	if firstItemIndex >= MAX_BLOG_ITEMS_NUMBER_PER_PAGE {
+		return fmt.Sprintf("%s%d", BLOG_ID_PREFIX, firstItemIndex-MAX_BLOG_ITEMS_NUMBER_PER_PAGE+1)
+	}
+	return ""
+}
+
+func buildPaginationURL(r *http.Request, page Page, id string) string {
+	return buildSourceOrigin(r.Host) + page.Route + "?from=" + id
+}
+
+func getPageNumberFromProductIndex(index int) int {
+	return (index / MAX_BLOG_ITEMS_NUMBER_PER_PAGE) + 1
+}
+
+func getBlogEntryIndexFromID(id string, blogItems []BlogItem) int {
+	blogEntryId, error := strconv.Atoi(id)
+	//default to the first page
+	if error != nil || blogEntryId > len(blogItems) {
+		return 0
```

Comment:
Because I am returning the index, not the id.
So for item 1, the ID would be post1 and the index would be 0.